### ISSUE TITLE
Generic reproducibility test

### ISSUE THREAD

**Issue by matthieu-vergne** [jMetal/jMetal#156]
considering that issue #155 is about reproducibility, which is a general property that every algorithm should be able to respect, I recommend to have a general test which checks all the algorithms, and which passes only when reproducibility is confirmed everywhere (and fails by telling which algorithm is not reproducible).

My way to do this kind of tests would be by having an abstract test class which contains the generic test and by requiring a method to initialise the algorithm. Maybe also another method to compare the runnings, if we cannot establish a generic way to confirm whether or not two runs have been identical. The problem is that it does not split well in several classes (you cannot extend more than one class in Java).

Another way could be to establish an interface and an implementation available for reuse, then for the concerned classes to implement the interface and to reuse the default implementation. But maybe a better solution would be to use [default methods](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html) introduced by Java 8 (another explanation [here](https://blog.idrsolutions.com/2015/01/java-8-default-methods-explained-5-minutes/)).

**Comment by matthieu-vergne** [comment]
> The problem is that it does not split well in several classes (you cannot extend more than one class in Java).

Problem fixed with Java 8, which allows to use interfaces with default methods. Because one class can implement several interfaces, we can merge generic tests without restrictions.

But for Junit to consider them, we need to use [Junit 5](http://junit.org/junit5/). I have tried it in another project of mine and I was able to create tests with such default methods. More investigation might be needed, but so far it works.
